### PR TITLE
Added option to instantly clear spawned particles when stopping particlefx

### LINKED ISSUE #5265: Ability to instantly clear existing particles
**Is your feature request related to a problem? Please describe (REQUIRED):**

Stopping a particlefx component ceases the production of new particles, but does not immediately delete the existing particles. This is not an issue in a lot of cases, e.g. when spawned particles have a short life span. However, where particles have a long life span (e.g. 10 seconds), they can remain visible for a long time after the particlefx component has been stopped.

**Describe the solution you'd like (REQUIRED):**

I would like to see functions like **gui.clear_particlefx()** and **particlefx.clear()** that instantly delete all existing particles in a particlefx component.

These should be distinct from **gui.stop_particlefx()** and **particlefx.stop()**, which merely stop the spawning of new particles rather than clearing existing ones.

Alternatively, an optional parameter could be added to the existing functions, e.g. **gui.stop_particlefx(node, clear_existing)** and **particlefx.stop(url, clear_existing)**.

**Describe alternatives you've considered (REQUIRED):**

I understand [unloading the collection containing the particlefx](https://forum.defold.com/t/delete-living-particles/64303) should instantly delete them. This is only useful in some situations.

A use case for me is where I have a gui particlefx component used as a backdrop for different types of announcements. In between each announcement I stop the particlefx component, then swap it to the next kind relevant to the next announcement. The effect is that particles from the old type still remain during the new announcement. This is illustrated in the gif attached:

ezgif-3-481adfcdcd70

### PR DESCRIPTION
**Pull request by britzl**
When `particlefx.stop()` is called all emitters immediately stop spawning new particles but already spawned particles will continue to live until they reach their end of life. Clearing/removing already spawned particles is only possible by either unloading the collection the particles belong to or by setting the tint alpha in the particlefx material to 0.0. 

To give developers a better alternative to clearing spawned particles It is now possible to specify this when the particle effect is stopped:

```
particlefx.stop(url, { clear = true })
gui.stop_particlefx(node, { clear = true })
```

Fixes #5265

### CODE REVIEW

File: engine/gamesys/src/gamesys/scripts/script_particlefx.cpp [hunk 1]
```diff
@@ -227,35 +227,57 @@ namespace dmGameSystem
...
         dmScript::ResolveURL(L, 1, &receiver, &sender);
 
+        int clear_particles = 0;
+        if (!lua_isnil(L, 2)) {
+            luaL_checktype(L, 2, LUA_TTABLE);
+            lua_pushvalue(L, 2);
+            lua_pushnil(L);
+            while (lua_next(L, -2)) {
+                const char* option = lua_tostring(L, -2);
+                if (strcmp(option, "clear") == 0)
```
Comment (britzl) [inline engine/gamesys/src/gamesys/scripts/script_particlefx.cpp:265]: Need help with option name:

* clear | clear_particles
* delete | delete_particles
* remove | remove_particles

?

Comment (AGulev) [reply engine/gamesys/src/gamesys/scripts/script_particlefx.cpp:265]: `clear` sounds good to me

Comment (AGulev) [reply engine/gamesys/src/gamesys/scripts/script_particlefx.cpp:265]: maybe `immediate_clear` if it's not too long

**Review by AGulev — APPROVED** [review]
👍 one of old bugs we finally solved!
I remember I reported it in October 2016 ([1](https://forum.defold.com/t/disable-particlefx-component-def-2267/3347?u=agulev) and [2](https://forum.defold.com/t/particle-end-event-like-animation-done-solved/709/14?u=agulev) ) when I worked on my very first game on Defold

File: engine/gamesys/src/gamesys/scripts/script_particlefx.cpp (hunk 1 above)
Comment (britzl) [reply engine/gamesys/src/gamesys/scripts/script_particlefx.cpp:265]: Thanks. I think I'll stick with ’clear’. ’immediate_clear’ is too long I think.

**Pull request merged**